### DISCUSSION TITLE
How to handle `FfiException`?

### DISCUSSION THREAD

**Discussion by justinmoon** [fzyzcjy/flutter_rust_bridge#490] [General]
Thanks for the great library.

My rust function returns a `Result`. I'm just printing out a raw `anyhow` error to the screen right now. This is how it looks: `FfiException(RESULT_ERROR, <anyhow message>, null)`. I spent 10 minutes searching around for any information on better ways to handle errors like this on the dart side and didn't find much.

Any suggestions?

**Comment by fzyzcjy** [comment]
Dart language has try-catch

```
try {
await your_rust_function();
}catch(e){
print('error!!!');
}
```

**Reply by justinmoon** [reply]
Thanks. How would I get just the `<anyhow message>` from my example? Would it be possible to do more granular error handling based on which error was returned by rust?

**Reply by fzyzcjy** [reply]
https://pub.dev/documentation/flutter_rust_bridge/latest/flutter_rust_bridge/FfiException-class.html

`e.code, e.details, e.message`, etc
